core/src/main/java/org/elasticsearch/index/query/TermsQueryBuilder.java

```diff
@@ -158,76 +287,127 @@ public TermsQueryBuilder disableCoord(boolean disableCoord) {
         return this;
     }
 
+    public boolean disableCoord() {
+        return this.disableCoord;
+    }
+
+    public boolean isTermsLookupQuery() {
+        return this.termsLookup.id != null;
```

Comment:
the problem comes in the setters in which a new object has to be created if null.